From 6f92d05a7aa6535d8fc8f1cb9432d85ff269a23a Mon Sep 17 00:00:00 2001
From: Diogo Ferreira <defer@cyngn.com>
Date: Wed, 7 Sep 2016 18:54:26 +0100
Subject: [PATCH 4/5] reference-ril: Downgrade IPV6 pdps to IPV4

The emulator RIL can't handle IPV6 emulation and many of the APNs that
get assigned to the emulator are in fact IPV6. This patch makes the
emulator RIL downgrade them automatically so that a successful data call
is always established, irrespective othe pdp type.

Change-Id: I5c5c682ac448ab787b74c30f64660a97e401ab72
---
 reference-ril/reference-ril.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/reference-ril/reference-ril.c b/reference-ril/reference-ril.c
index 272b7b2..7ef0a88 100644
--- a/reference-ril/reference-ril.c
+++ b/reference-ril/reference-ril.c
@@ -1728,6 +1728,11 @@ static void requestSetupDataCall(void *data, size_t datalen, RIL_Token t)
             pdp_type = "IP";
         }
 
+        if (!strcmp("IPV6", pdp_type)) {
+            RLOGW("IPV6 pdp requested, but reference ril only supports IPV4, downgrading.");
+            pdp_type = "IP";
+        }
+
         asprintf(&cmd, "AT+CGDCONT=1,\"%s\",\"%s\",,0,0", pdp_type, apn);
         //FIXME check for error here
         err = at_send_command(cmd, NULL);
-- 
2.7.4

